search-app/src/cmr/search/api/routes.clj

```diff
@@ -315,20 +305,46 @@
      :headers {common-routes/CONTENT_TYPE_HEADER (mt/with-utf-8 mt/json)}
      :body results}))
 
-(def robots-txt-response
-  "Returns the robots.txt response."
-  {:status 200
-   :body (slurp (io/resource "public/robots.txt"))})
+(defn find-query-str-mixed-arity-param
+  "Return the first parameter that has mixed arity, i.e., appears with both single and multivalued in
+  the query string. e.g. foo=1&foo[bar]=2 is mixed arity, so is foo[]=1&foo[bar]=2. foo=1&foo[]=2 is
+  not. Parameter with mixed arity will be flagged as invalid later."
+  [query-str]
+  (when query-str
+    (let [query-str (-> query-str
+                        (str/replace #"%5B" "[")
+                        (str/replace #"%5D" "]")
+                        (str/replace #"\[\]" ""))]
+      (last (some #(re-find % query-str)
+                  [#"(^|&)(.*?)=.*?\2\["
+                   #"(^|&)(.*?)\[.*?\2="])))))
+
+(defn mixed-arity-param-handler
+  "Detect query string with mixed arity and throws a 400 error. Mixed arity param is when a single
+  value param is mixed with multivalue. One specific case of this is for improperly expressed options
+  in the query string, e.g., granule_ur=*&granule_ur[pattern]=true. Ring parameter handling throws
+  500 error when it happens. This middleware handler returns a 400 error early to avoid the 500 error
+  from Ring."
+  [f]
+  (fn [request]
+    (when-let [mixed-param (find-query-str-mixed-arity-param (:query-string request))]
+      (svc-errors/throw-service-errors
+        :bad-request
+        [(msg/mixed-arity-parameter-msg mixed-param)]))
+    (f request)))
+
+(defn default-error-format-fn
```

Comment:
Did some more poking around to see how else CMR handles error formatting, and these two examples are the only ones I could see that don't use the default.

Still, that being said, since it does occur in two places, it might make sense to briefly ponder the best home for this function ... maybe some place in common?